### DISCUSSION TITLE
"Support for password authentication was removed" error

### DISCUSSION THREAD

**Discussion by john-science** [JamesIves/github-pages-deploy-action#858] [Q&A] [answered]
I am getting the error `Support for password authentication was removed` error. But I am using a token, and have cranked it up to have fully permissions.

I am trying to use GH Actions to auto-gen my docs in ([this repo](https://github.com/john-science/armi/)), then PR them to the documentation website repo [here](https://github.com/john-science/terrapower.github.io).

The relevant portion of [my YAML](https://github.com/john-science/armi/blob/test_pages/.github/workflows/docs.yaml):

```yaml
      - name: deploy
        uses: JamesIves/github-pages-deploy-action@4.0.0
        with:
          token: ${{ secrets.ACCESS_TOKEN }}
          repository-name: john-science/terrapower.github.io
          commit-message: "${{ github.repository }}@${{ github.sha }}"
          branch: master
          folder: doc/_build/html
          target-folder: armi
```

My full error log is [here](https://github.com/john-science/armi/runs/3497869861?check_suite_focus=true#step:9:52), but in short the error I get:

```bash
/usr/bin/git push --force ***github.com/john-science/terrapower.github.io.git github-pages-deploy-action/qzoaqahrc:master
remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
remote: Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
fatal: Authentication failed for 'https://github.com/john-science/terrapower.github.io.git/'
```

Any help would be appreciated. I must have a hole in my understanding, because I don't grok the error message at all.

**Comment by john-science** [comment, marked as answer]
The issue appears to be a bad/faulty/misleading error message in the GitHub token API.  

Explanation here:  https://github.com/JamesIves/github-pages-deploy-action/issues/859

**Comment by TheArrowsmith** [comment]
If anyone else is coming here from Google... I got this same error message while trying to get started with a similar action, https://github.com/peaceiris/actions-gh-pages

My problem was that I'd written this in my workflow config file:

          github_token: ${secrets.GITHUB_TOKEN}

But it actually needed to use double braces `{{`/`}}`, not single `{`/`}`:

          github_token: ${{secrets.GITHUB_TOKEN}}

As @john-science noted above, the error message from GitHub is very misleading.
